File: ethereum/src/eth-py/information/get_balance.py
import os
from web3 import Web3
from dotenv import load_dotenv

def get_balance():
    load_dotenv()
    provider_url = os.getenv('ENDPOINT_TESTNET')
    address = os.getenv('PUBLIC_KEY_SENDER')

    web3 = Web3(Web3.HTTPProvider(provider_url))
    balance = web3.eth.get_balance(address)
    
    print(f"Balance: {web3.fromWei(balance, 'ether')} ETH")

File: ethereum/src/eth-rs/src/information/get_balance.rs
use ethers::prelude::*;
use std::convert::TryFrom;
use dotenv::dotenv;
use std::env;

pub async fn get_balance() -> Result<(), Box<dyn std::error::Error>> {
    // .envファイルを読み込む
    dotenv().ok();

    // 環境変数からEthereumアドレスとプロバイダーURLを取得
    let address = env::var("PUBLIC_KEY_SENDER")?;
    let provider_url = env::var("ENDPOINT_TESTNET")?;
    let address = address.parse::<Address>()?;

    // プロバイダを作成し、残高を取得
    let provider = Provider::<Http>::try_from(provider_url)?;
    let balance = provider.get_balance(address, None).await?;
    
    // 残高を表示
    println!("Balance: {:?}", balance);
    Ok(())
}
